tests/test_band_masks.py

```diff
@@ -150,10 +150,10 @@ def test_masking_sidecar_mask(tiffs):
     # If the dataset has a .msk sidecar mask band file, all masks will
     # be derived from that file.
     with rasterio.open(str(tiffs.join('sidecar-masked.tif'))) as src:
-        for flags in src.mask_flags:
-            assert flags & MaskFlags.per_dataset
-            assert not flags & MaskFlags.alpha
-            assert not flags & MaskFlags.nodata
+        for flags in src.mask_flag_enums:
+            assert MaskFlags.per_dataset in flags
+            assert MaskFlags.alpha not in flags
+            assert MaskFlags.nodata not in flags
         rgb = src.read(masked=True)
         assert rgb.mask.all()
         r = src.read(1, masked=True)
```

Comment:
Can we add a test for an RGBA tif? That's a good case to test since there's more than a single flag per band and the flags differ by band

```
# tests/data/alpha.tif
[
  ["per_dataset", "alpha"], 
  ["per_dataset", "alpha"], 
  ["per_dataset", "alpha"], 
  ["all_valid"]
]
```